nucliadb_utils/nucliadb_utils/signals.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import asyncio
+from collections.abc import Awaitable
+from enum import Enum
+from functools import partial, wraps
+from inspect import iscoroutinefunction
+from typing import Any, Callable, Type, Union
+
+
+class ListenerPriority(Enum):
+    DONT_CARE = 0
+    CRITICAL = 1
+
+
+class Signal:
```

Comment:
Can you add some docstring here to explain what this util class can be used for?

Reply:
also, can you add unit tests to this module? it would help on understanding its usage too